Test failures with Python 3.12

Hello in Fedora, I try to build pythran 0.13.1 with https://github.com/serge-sans-paille/pythran/pull/2119 and https://github.com/serge-sans-paille/pythran/pull/2120 applied on top using Python 3.12.0b3. That is https://src.fedoraproject.org/rpms/pythran/pull-request/25

I repeatedly get this failure:

```
=================================== FAILURES ===================================
__________________________ TestDoctest.test_toolchain __________________________
[gw5] linux -- Python 3.12.0 /usr/bin/python3
self = <pythran.tests.test_xdoc.TestDoctest testMethod=test_toolchain>
>   lambda self: generic_test_package(self, module))
pythran/tests/test_xdoc.py:89: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/test_xdoc.py:83: in generic_test_package
    self.assertEqual(failed, 0)
E   AssertionError: 1 != 0
----------------------------- Captured stdout call -----------------------------
**********************************************************************
File "/builddir/build/BUILD/pythran-0.13.1/pythran/toolchain.py", line 465, in pythran.toolchain.compile_pythranfile
Failed example:
    cpp_path = compile_pythranfile('pythran_test.py', cpponly=True)
Expected nothing
Got:
    No pythran specification, nothing will be exported
**********************************************************************
1 items had failures:
   1 of   7 in pythran.toolchain.compile_pythranfile
***Test Failed*** 1 failures.
_____________________________ TestDoctest.test_cli _____________________________
[gw0] linux -- Python 3.12.0 /usr/bin/python3
self = <pythran.tests.test_xdoc.TestDoctest testMethod=test_cli>
    @pytest.mark.skipif(sys.platform == "win32",
                        reason="We should create a file for windows.")
    def test_cli(self):
        tmpfile = self.adapt_rst('../../docs/CLI.rst')
        failed, _ = doctest.testfile(tmpfile, False)
>       self.assertEqual(failed, 0)
E       AssertionError: 1 != 0
pythran/tests/test_xdoc.py:40: AssertionError
----------------------------- Captured stdout call -----------------------------
**********************************************************************
File "/tmp/tmp69kfq1wu", line 68, in tmp69kfq1wu
Failed example:
    import subprocess ; res = subprocess.check_output(" /usr/bin/python3 -m pythran.run -e cli_bar.py -o cli_bar.hpp", shell=True).decode("ascii").strip() ; print(res, end="")
Expected nothing
Got:
    No pythran specification, nothing will be exported
**********************************************************************
1 items had failures:
   1 of  25 in tmp69kfq1wu
***Test Failed*** 1 failures.
----------------------------- Captured stderr call -----------------------------
<frozen runpy>:128: RuntimeWarning: 'pythran.config' found in sys.modules after import of package 'pythran', but prior to execution of 'pythran.config'; this may result in unpredictable behaviour
=============================== warnings summary ===============================
...
=========================== short test summary info ============================
FAILED pythran/tests/test_xdoc.py::TestDoctest::test_toolchain - AssertionErr...
FAILED pythran/tests/test_xdoc.py::TestDoctest::test_cli - AssertionError: 1 ...
==== 2 failed, 3564 passed, 56 skipped, 147 warnings in 3733.00s (1:02:13) =====
```

Do you have an idea what is going on here?

--- a/docs/CLI.rst
+++ b/docs/CLI.rst
@@ -65,7 +65,7 @@ That's some heavily templated code ;-) Pythran can then compile it for you to a 
 Pythran can also generate raw C++ code, using the ``-e`` switch::
 
   $> printf 'msg = \"hello world\"\ndef bar(): print(msg)' > cli_bar.py
-  $> pythran -e cli_bar.py -o cli_bar.hpp
+  $> pythran -e cli_bar.py -w -o cli_bar.hpp
   $> printf '#include \"cli_bar.hpp\"\nusing namespace __pythran_cli_bar ; int main() { bar()(); return 0 ; }' > cli_bar.cpp
   $> `pythran-config --compiler --cflags` -std=c++11 cli_bar.cpp -o cli_bar
   $> ./cli_bar

--- a/pythran/spec.py
+++ b/pythran/spec.py
@@ -90,9 +90,9 @@ class Spec(object):
                 self.functions[fname] = (signatures,)
 
         if not self:
-            import logging
-            logging.warning("No pythran specification, "
-                            "nothing will be exported")
+            from pythran.log import logger
+            logger.warning("No pythran specification, "
+                           "nothing will be exported")
 
     def keys(self):
         return list(self.functions.keys()) + list(self.capsules.keys())

